Add FreeBSD support

akhramov opened the pull request:
The library supports Linux, OSX and Windows, but not FreeBSD yet.

This change
* Adds FreeBSD support via `ptrace (2)` `PT_VM_ENTRY` call.
* Bumps `libc` crate version in order to get broader `ptrace (2)` call
  types support, namely `PT_VM_ENTRY`.
* Adds Cirrus-backed FreeBSD CI. `libc` crate uses Cirrus for the very
  same purpose.
* Adds mentions of FreeBSD in docstrings where appropriate.

akhramov committed:
Add FreeBSD support

The library supports Linux, OSX and Windows, but not FreeBSD yet.

This change
* Adds FreeBSD support via `ptrace (2)` `PT_VM_ENTRY` call.
* Bumps `libc` crate version in order to get broader `ptrace (2)` call
  types support, namely `PT_VM_ENTRY`.
* Adds Cirrus-backed FreeBSD CI. `libc` crate uses Cirrus for the very
  same purpose.
* Adds mentions of FreeBSD in docstrings where appropriate.

akhramov commented:
In order to set Cirrus CI up, please follow the guide: https://cirrus-ci.org/guide/quick-start/

I'd be happy to assist.

benfred commented on src/freebsd_maps/mod.rs:
```diff
@@ -0,0 +1,161 @@
+                Some(entry)
+            }
+            _ => {
+                ptrace::detach(pid).ok()?;
```
Having the detach call here means that if the iterator isn't ever completely exhausted, the target process will remain locked.

Can we add to a 'Drop' trait instead?  (something like: https://github.com/benfred/py-spy/blob/1717b91d7fafc6910e1e9b76e3397a7d58e883dc/remoteprocess/src/linux/mod.rs#L170)

akhramov replied:
Good catch, fixed!

akhramov committed:
Add FreeBSD support

This change contains post-code review fixes, namely:

* implement `Drop` trait for `VmEntryIterator` in order to avoid
  possible error when the iterator is not exhausted and the tracee
  process remain attached.
* Move process locking logic to `VmEntryIterator::new`.

benfred commented:
Thanks for this! I've pushed a new version that includes these changes (0.1.6). Will get CI set up in a bit,

akhramov commented:
Alright, I have updated the py-spy PR accordingly, thanks!
